### PR TITLE
Allows findWidget to use all given findOptions and adds clear to widget object

### PR DESCRIPTION
**Pull request by ryan-bradford**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [X] Tests for the changes have been added (for bug fixes / features)
-   [ ] Examples have been added / updated (for bug fixes / features)
-   [X] Changelog has been updated

## PR Type

What kind of change does this PR introduce?

-   [X] Bugfix
-   [X] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Example website changes
-   [ ] Version bump
-   [ ] Other... Please describe:

## What does this change do?
The widget finder was ignoring most of the arguments given to `findOptions` in a call to `findWidget`. This meant we couldn't find widgets by text or index. This patches the bug.

Also, this adds tests for the cypress widget objects. It does this by mocking the cy object and making assertions about calls to it.

Furthermore, clear was not a method that was callable from inside a widget object, even though it should have been. This adds clear as a method to a widget object.

## What manual testing did you do?
1. Ran the data exporter tests and observed them passing.

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [X] No

### CODE REVIEW

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts [hunk 1]
```diff
@@ -40,11 +40,23 @@ export class CypressWidgetObjectFinder<T> {
         if (findOptions?.cssSelector) {
             tagName += `${findOptions.cssSelector}`;
```
Comment (xvu-vmware) [inline projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts:41]: Don't you need to prefix with a space if you're going to concat the two selectors?

For example:
```tagName += ` ${findOptions.cssSelector}`;```

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts [hunk 2]
```diff
@@ -40,11 +40,23 @@ export class CypressWidgetObjectFinder<T> {
...
         }
+
         const id = idGenerator.generate();
-        const search = findOptions?.ancestor
-            ? cy.get(findOptions.ancestor).find(tagName)
-            : cy.get(tagName, findOptions?.options);
-        const root = search.as(id);
+
+        const ancestor = findOptions?.ancestor ? cy.get(findOptions?.ancestor) : cy.get('body');
```
Comment (xvu-vmware) [inline projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts:46]: We're not using the `FindCypressWidgetOptions.options` for timeout.

For example:
```
// This code isn't complete. Don't pass in `timeout` if it's not defined.
const timeout = findOptions?.options?.timeout;
const ancestor = findOptions?.ancestor ? cy.get(findOptions?.ancestor, { timeout } ) : cy.get('body',  { timeout });
```

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts [hunk 3]
```diff
@@ -40,11 +40,23 @@ export class CypressWidgetObjectFinder<T> {
...
+        const parentWidget = new CypressWidgetObjectElement(ancestor, false, undefined);
+        let query = widgetConstructor.tagName;
+        if (findOptions?.cssSelector) {
+            query = query + findOptions.cssSelector;
```
Comment (xvu-vmware) [inline projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts:50]: Don't you need a space when combining selectors?

**Review by xvu-vmware — APPROVED** [review]

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts (hunk 3 above)
Comment (juanmendes) [reply projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts:50]: No, the selector should be applied to the root element, it's combining it with the tag name? I believe, @ryan-bradford ?

**Comment by juanmendes** [comment]
It doesn't make sense to search for a widget by text, right? Can we remove that option?

**Comment by xvu-vmware** [comment]
> It doesn't make sense to search for a widget by text, right? Can we remove that option?

Imagine I'm on the vApp card screen and I would like to perform power on action for "vApp-1". I want to call `findWidget()` to get the vApp card widget that has the vApp name of "vApp-1". I think it makes sense to be able to find by text.

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts (hunk 3 above)
Comment (ryan-bradford) [reply projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts:50]: Yea, there's a case where you want to get some tagName with a specific class applied (`vcd-data-exporter.vms-exporter`), and for this case, we can't have a space.

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts (hunk 1 above)
Comment (ryan-bradford) [reply projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts:41]: Same as in the below discussion.

File: projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts (hunk 2 above)
Comment (ryan-bradford) [reply projects/components/src/utils/test/widget-object/cypress/cypress-widget-finder.ts:46]: Updated to include this!

**Pull request merged**